action.yaml

```diff
@@ -17,14 +17,16 @@ inputs:
   token:
     description: 'Github token'
     default: ${{ github.token }}
+    required: false
```

Comment:
This is required.

Reply:
This is not being taken from the user. If I mark it required, user will have to give it as an input.

Reply:
No. It's an implication that it has to be provided for it to work. If the default value is not available at some point the action would fail before making the call.

Reply:
Okay. I'll add a check for it. But I believe the error message should be different in this case, something like: `Could not extract GITHUB_TOKEN` or `Could not find GITHUB_TOKEN`?

Reply:
Just GitHub token is unset or something. User can always update it although you're "hiding" it.

Reply:
Already added a commit

Reply:
You should use just the word `token` as that's what the user input is. What do you think?

Reply:
Actually user is not aware of this input. And if because of some reason this input is null, we should clearly state that GITHUB_TOKEN could not be found. Just using the word token in error message would not explain the error.

Reply:
I'm talking about the case where the user saw the action.yml instead of docs

Reply:
According to me, the use of `token` is supposed to be an internal implementation for getting the GITHUB_TOKEN. Exposing the variable in error logs doesn't seem right, irrespective of whether user has seen the action.yml or not. If there was another way of getting the GITHUB_TOKEN in code, I wouldn't have introduced that input. 😛 
But if `token` seems more suitable for the error message, I'll update it. What do you think?

Reply:
Maybe cover both. 
` 'token' input is not supplied. Set it to a PAT/GITHUB_TOKEN`

Reply:
Sounds good.